aya/src/bpf.rs

```diff
@@ -224,6 +228,62 @@ impl<'a> EbpfLoader<'a> {
         self
     }
 
+    /// Allows programs containing unsupported maps for the current kernel to be loaded
+    /// by skipping map creation and relocation before loading.
+    ///
+    /// This is useful when you have a single ebpf program containing e.g. a `RingBuf`
+    /// and a `PerfEventArray` and you decide which one to use before loading the bytecode.
+    ///
+    /// Must be used with `.set_global()` to signal if the map is supported in the ebpf program.
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// use aya::EbpfLoader;
+    /// use aya_obj::generated::bpf_map_type;
+    /// use std::collections::HashSet;
+    ///
+    /// let ringbuf_supported = 0;
+    /// let mut set = HashSet::new();
+    /// set.insert(bpf_map_type::BPF_MAP_TYPE_RINGBUF);
+    /// let ebpf = EbpfLoader::new()
+    ///     .ignore_maps_by_type(set)
+    ///     .set_global("RINGBUF_SUPPORTED", &ringbuf_supported, true)
+    ///     .load_file("file.o")?;
+    /// # Ok::<(), aya::EbpfError>(())
+    /// ```
+    ///
+    pub fn ignore_maps_by_type(&mut self, set: HashSet<bpf_map_type>) -> &mut Self {
```

Comment:
We shouldn't expose `bpf_map_type` here. It's a type that we don't directly
control so it'll break our API. I don't think we already have our own MapType,
maybe it's time to have it.

Reply:
Do you think the function should be removed until we have our own MapType, or do you want me to introduce it?

I can remove it for this PR and create a new PR that introduces MapType if you think that's best?

Reply:
Up to you I'm fine with either!

Reply:
Removed by_type, that can come in a later PR if someone needs it